cray_product_catalog/catalog_update.py

```diff
@@ -134,6 +136,19 @@ def active_field_exists(product_data):
     return any("active" in product_data[version] for version in product_data)
 
 
+def create_config_map(api_instance, name, namespace):
+    """Create new product ConfigMap."""
+    try:
+        new_cm = V1ConfigMap()
+        new_cm.metadata = V1ObjectMeta(name=name)
+        api_instance.create_namespaced_config_map(
+            namespace=namespace, body=new_cm
+        )
+        LOGGER.debug("Created product ConfigMap %s/%s", namespace, name)
+    except ApiException as err:
+        LOGGER.exception("Error calling create_namespaced_config_map")
```

Comment:
The `err` variable is unused here. I think it would be preferable to just log the string representation of the exception rather than the full stack trace. I don't think the end user is going to care what the stack trace is. Logging stack traces can even be a security vulnerability depending on the context.

I also wonder if `warning` is a more appropriate level. This situation is not necessarily fatal, and this could potentially be logged many times by the calling loop in `update_config_map`. Ultimately, it would be nice if that loop called a function to either read the configmap or create it, as appropriate. Then it would be nice if that function either returned the configmap data that it read or created, or raised a custom exception if there was a non-fatal failure for which we want to retry. This would probably clean up the loop and the logging of warnings significantly.

Reply:
Implementated to exit update_config_map if create_config_map failes